Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1296,35 +1286,32 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
      * @param namespace  the namespace where the document lives.
      * @param documentId the _id of the document.
      */
-    private func deleteOneFromRemote(
-        namespace: MongoNamespace,
-        documentId: BSONValue
-        ) throws {
-        guard let lock = syncConfig[namespace]?.nsLock else {
+    private func deleteOneFromRemote(namespace: MongoNamespace, documentId: BSONValue) throws {
+        guard let nsConfig = syncConfig[namespace] else {
             return
         }
 
         var deferredEventEmittingBlock: DeferredBlock? = nil
```

Comment:
[nit] with this new locking scheme, we don't need the deferred event emitting block. Since we're not deferring to unlock the lock, we can just emit the event at the end of the function